dev/release/download_rc_binaries.py

```diff
@@ -121,17 +121,28 @@ def _download_url(self, url, dest_path, *, extra_args=None):
             dest_path,
             url,
         ]
-        proc = subprocess.Popen(cmd, stdout=subprocess.PIPE,
-                                stderr=subprocess.PIPE)
-        stdout, stderr = proc.communicate()
-        if proc.returncode != 0:
-            try:
-                # Don't leave possibly partial file around
-                os.remove(dest_path)
-            except IOError:
-                pass
-            raise Exception(f"Downloading {url} failed\n"
-                            f"stdout: {stdout}\nstderr: {stderr}")
+        # Retry subprocess in case it fails with OpenSSL Connection errors
+        # https://issues.apache.org/jira/browse/INFRA-25274
+        for attempt in range(5):
+            if delay := attempt * 3:
```

Comment:
We may want to avoid `:=` because it's available since Python 3.8.
I know that Python 3.7 reached EOL but some our (old) environments for verification may use Python 3.7 or earlier...

Reply:
```suggestion
            delay = attempt * 3
            if delay:
```
Thanks @kou ! Is this ok?